libs/wss/src/compression/byte.rs

```diff
@@ -0,0 +1,137 @@
+#[allow(clippy::excessive_precision)]
+const LN_10: f64 = 2.302_585_092_994_045_684_084_0;
+#[allow(clippy::excessive_precision)]
+const LOG2: f64 = 1.442_695_040_888_963_407_0;
+
+#[allow(clippy::suboptimal_flops)]
+pub fn decompress(data: &[u8]) -> Vec<i16> {
+    let magic_number: f64 = (LN_10 * LOG2) / 28.125_740_425_151_72;
+    let mut output_data = Vec::with_capacity(data.len());
+    let mut last_val: i16 = 0;
```

Comment:
Again this is the mono only. Stereo data has two delta's.
Also delta8 compression is very similar to delta4.
Except that the lookup table has 255 values, where delta4 has 15.

Why is this code so much different 🤔

Reply:
A good test might be
A3/Sounds_F_Enoch/Assets/Environment/Backgrounds/Meadow/Meadow_Bg_Day_Quiet
Our encoder used to mess up on that until last update.

> Bis optionally use Delta8 (byte) or Delta4 (nibble) 'Lossy' Compression for mono pcm data only.
> The facility exists to handle stereo compression but it isn't employed. 

That may have been right when this was written 15 years ago. it isn't anymore.